server/app/db/crud/bot_types.py

```diff
@@ -0,0 +1,206 @@
+
+    # creating query
+    q = MySQLQuery.from_(bot_types).select("*").where(bot_types.name == name)
+
+    log_query(str(q))
+
+    try:
+        await con.execute(str(q))
+        row = await con.fetchone()
+        # TODO: Don't know if this will throw an error or return a empty
+        # response, need to check it
+        if len(row) == 0:
+            return None, RecordNotFound
+        resp = bot_type_schema.create_BotTypeInDB_from_tuple(row)
+        logging.debug(f"for the data={resp.dict()} while fetching bot with {name=}")
+        logging.info(f"Successfully fetched the bot with {name=}")
+        return resp, None
+    except Exception as e:
+        logging.error(f"Unable to query with {name=} with query={q} due to `{e}`")
+        return None, e
+
+
+async def get_bot_type_with_given_id(
+    con: Cursor, id: int
+) -> Tuple[Union[bot_type_schema.BotTypeInDB, None], Union[Exception, None]]:
+    """Fetches the bot with the given id"""
+    logging.info(f"Trying the fetch bot_type with {id=}")
+
+    # creating query
+    q = MySQLQuery.from_(bot_types).select("*").where(bot_types.id == id)
+
+    log_query(str(q))
+
+    # executing the query
+    try:
+        await con.execute(str(q))
+        row = await con.fetchone()
+        # TODO: Don't know if this will throw an error or return a empty
+        # response, need to check it
+        if len(row) == 0:
+            return None, RecordNotFound
+        resp = bot_type_schema.create_BotTypeInDB_from_tuple(row)
+        logging.debug(f"for the data={resp.dict()} while fetching bot with {id=}")
+        logging.info(f"Successfully fetched the bot with the {id=}")
+        return resp, None
+    except Exception as e:
+        logging.error(f"Unable to query with {id=} with query={q} due to `{e}`")
+        return None, e
+
+
+async def update_bot_name(
+    con: Cursor, id: int, name: str
+) -> Tuple[bool, Union[Exception, None]]:
+    """Changes the name of the the bot with the given id"""
+
+    logging.info(f"Trying to the name of the bot with {id=}, to {name}")
+
+    # TODO: Add data validation
+
+    # creating query
+    q = MySQLQuery.update(bot_types).set(bot_types.name, name).where(bot_types.id == id)
+
+    log_query(str(q))
+
+    try:
+        await con.execute(str(q))
+        await con.commit()
+        logging.info(f"Successfully updated bot_type name to `{name}` with {id=}")
+        return True, None
+    except Exception as e:
+        # TODO: Need to handle 1. invalid id, 2. duplicate name
+        logging.error(
+            f"Unable to update bot_type with {id=} to {name=} with query=`{q}`"
+            + f"due to `{e}`"
+        )
+        return False, e
+
+
+async def delete_bot_type_with_given_id(
+    con: Cursor, id: int
+) -> Tuple[bool, Union[Exception, None]]:
+    """deletes the bots with the given id"""
+
+    logging.info(f"Trying to delete bot with {id=}")
+
+    q = MySQLQuery.from_(bot_types).delete().where(bot_types.id == id)
```

Comment:
Shouldn't we clear all bots from bots table before deleting from bot_types?

Reply:
it will cascade delete

Reply:
Ahh right. But Ig we have to find a safe way to delete bots so that stocks don't get stranded in any of the bot accounts